sqlmesh/core/console.py

```diff
@@ -912,6 +912,9 @@ def _show_missing_dates(self, plan: Plan, default_catalog: t.Optional[str]) -> N
             backfill = self._limit_model_names(backfill, self.verbose)
         self._print(backfill)
 
+    def _validate_date_input(self, date_str: str) -> None:
+        self._print(f"Entered value is calculated as '{to_datetime(date_str)}'")
```

Comment:
I don't think this is correct. If the entered input is a date (eg. `2024-01-01`) then SQLMesh will treat it as such and not as a datetime object. This is especially relevant for end dates since dates are treated as inclusive and not exclusive (unlike timestamps, datetimes, etc).

So at the very least it should be something like `to_date(date_str) if is_date(date_str) else to_datetime(date_str)`.

Though I'm still not 100% sure why this is needed.